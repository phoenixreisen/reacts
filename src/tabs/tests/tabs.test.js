/**
 * @jest-environment jsdom
 */
 import {render, fireEvent, waitFor, screen} from '@testing-library/react';
 import renderer from 'react-test-renderer';
 import Tabs from '../tabs.r';
 import React from 'react';

 describe('Tabs should', () => {

     test('render correctly', () => {
         const component = renderer.create(
             <Tabs tabs={['Tab I', 'Tab II']}>
                 <div>Tab I</div>
                 <div>Tab II</div>
             </Tabs>
         );
         const snap = component.toJSON();
         expect(snap).toMatchSnapshot();
     });

     test('render single tab not as link', () => {
        const component = renderer.create(
            <Tabs tabs={['One & Only']}>
                <div>One & Only</div>
            </Tabs>
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
     });

     test('check that amount of tabs and contents is equal', () => {
        let error = false;
        try {
            const component = render.create(
                <Tabs tabs={['Tab I']}>
                    <div>Tab I</div>
                    <div>Tab II</div>
                </Tabs>
            );
        } catch(e) {
            error = true;
        }
        expect(error).toBe(true);
     });

     test('change on click', () => {
        let active = null;
        let inactive = null;

        const component = render(
            <Tabs tabs={['Tab I', 'Tab II']}>
                <div>Tab Content 1</div>
                <div>Tab Content 2</div>
            </Tabs>
        );
        active = document.querySelector('.is-visible-tab');
        inactive = document.querySelector('.is-hidden-tab');
        expect(active?.innerHTML.includes('Tab Content 1')).toBeTruthy();
        expect(inactive?.innerHTML.includes('Tab Content 2')).toBeTruthy();
        fireEvent.click(screen.getByText('Tab II'));
        active = document.querySelector('.is-visible-tab');
        inactive = document.querySelector('.is-hidden-tab')
        expect(active?.innerHTML.includes('Tab Content 2')).toBeTruthy();
        expect(inactive?.innerHTML.includes('Tab Content 1')).toBeTruthy();
        fireEvent.click(screen.getByText('Tab I'))
        active = document.querySelector('.is-visible-tab');
        inactive = document.querySelector('.is-hidden-tab')
        expect(active?.innerHTML.includes('Tab Content 1')).toBeTruthy();
        expect(inactive?.innerHTML.includes('Tab Content 2')).toBeTruthy();
    });
 });