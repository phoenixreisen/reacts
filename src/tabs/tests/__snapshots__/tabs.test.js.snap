// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly 1`] = `
<article
  className="tabs"
>
  <div
    className="tabs__tabs-bar"
  >
    <a
      className="tab tab-0 tab--link is-active-tab"
      href="#"
      onClick={[Function]}
    >
      Tab I
    </a>
    <a
      className="tab tab-1 tab--link is-inactive-tab"
      href="#"
      onClick={[Function]}
    >
      Tab II
    </a>
  </div>
  <div
    className="tabs__tabs-content"
  >
    <div
      className="tab-content tab-content-0 is-visible-tab"
    >
      <div>
        Tab I
      </div>
    </div>
    <div
      className="tab-content tab-content-1 is-hidden-tab"
    >
      <div>
        Tab II
      </div>
    </div>
  </div>
</article>
`;

exports[`Tabs should render single tab not as link 1`] = `
<article
  className="tabs"
>
  <div
    className="tabs__tabs-bar"
  >
    <a
      className="tab tab-0  is-active-tab"
      href="#"
      onClick={[Function]}
    >
      One & Only
    </a>
  </div>
  <div
    className="tabs__tabs-content"
  >
    <div
      className="tab-content tab-content-0 is-visible-tab"
    >
      <div>
        One & Only
      </div>
    </div>
  </div>
</article>
`;
